src/java/org/apache/cassandra/db/commitlog/SimpleCachedBufferPool.java

```diff
@@ -107,6 +120,9 @@ public void releaseBuffer(ByteBuffer buffer)
     public void shutdown()
     {
         bufferPool.clear();
+        for (FastThreadLocal<ByteBuffer> bbHolder : reusableBBHolder.values())
+            bbHolder.remove();
+        reusableBB.remove();
```

Comment:
`remove` is called twice on the same `ThreadLocal` instance as the `reusableBB` is also in the `bbHolder` variable

Moreover, the `remove` call will remove the buffers values associated to the thread calling the `shutdown` method and I do not believe that it is what we want. The goal of the approach is to reuse buffer. Calling `remove` will prevent the reuse on the next call forcing the buffer to be recreated.